packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2001,65 +2034,172 @@ export function writeEndSegment(
+//       p.s = 'l';
+//     }, (e) => {
+//       p.s = 'e';
+//       throw e;
+//     });
+//     return p;
+//   }
+
+//   return function insertStyle(styles) {
+//     let i = 0;
+//     let dependencies, style, href, precedence, attr, loadingState, resourceEl;
+
+//     while (style = styles[i++]) {
+//       let j = 0;
+//       href = style[j++];
+//       // We check if this resource is already in our resourceMap and reuse it if so.
+//       // If it is already loaded we don't return it as a depenendency since there is nothing
+//       // to wait for
+//       loadingState = resourceMap.get(href);
+//       if (loadingState) {
+//         if (loadingState.s !== 'l') {
+//           dependencies ? dependencies.push(loadingState) : dependencies = [loadingState];
 //         }
-//       } else if (
-//         data === SUSPENSE_START_DATA ||
-//         data === SUSPENSE_PENDING_START_DATA ||
-//         data === SUSPENSE_FALLBACK_START_DATA
-//       ) {
-//         depth++;
+//         continue;
+//       }
+
+//       // We construct our new resource element, looping over remaining attributes if any
+//       // setting them to the Element.
+//       resourceEl = thisDocument.createElement("link");
+//       resourceEl.href = href;
+//       resourceEl.rel = 'stylesheet';
+//       resourceEl.dataset.rprec = precedence = style[j++];
+//       while(attr = style[j++]) {
+//         resourceEl.setAttribute(attr, style[j++]);
+//       }
+
+//       // We stash a pending promise in our map by href which will resolve or reject
+//       // when the underlying resource loads or errors. We add it to the dependencies
+//       // array to be returned.
+//       loadingState = getPromise(resourceEl);
+//       resourceMap.set(href, loadingState);
+//       dependencies ? dependencies.push(loadingState) : dependencies = [loadingState];
+
+//       // The prior style resource is the last one placed at a given
+//       // precedence or the last resource itself which may be null.
+//       // We grab this value and then update the last resource for this
+//       // precedence to be the inserted element, updating the lastResource
+//       // pointer if needed.
+//       let prior = precedences.get(precedence) || lastResource;
+//       if (prior === lastResource) {
+//         lastResource = resourceEl
+//       }
+//       precedences.set(precedence, resourceEl)
+
+//       // Finally, we insert the newly constructed instance at an appropriate location
+//       // in the Document.
+//       if (prior) {
+//         prior.parentNode.insertBefore(resourceEl, prior.nextSibling);
+//       } else {
+//         let head = thisDocument.head;
+//         head.insertBefore(resourceEl, head.firstChild);
 //       }
 //     }
+//     return dependencies;
+//   }
+// })()
 //
-//     const nextNode = node.nextSibling;
-//     parentInstance.removeChild(node);
-//     node = nextNode;
-//   } while (node);
+// const completeBoundary = (function() {
```

Comment:
Relatively to the small code, these closures add quite a bit of code. In this case you can move flipBoundary into the plain completeBoundary so you don't need a closure.

It allocates once per complete call instead of once, but these will be very few calls and not long lived so it's not worth optimizing for. It creates an extra environment which isn't great neither.

Reply:
ah that is clever...